flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTable.java

```diff
@@ -57,12 +60,15 @@
 
 	private final SqlNodeList propertyList;
 
+	@Nullable
```

Comment:
Why `primaryKeyList` and `uniqueKeysList` are still nullable? 
I think we have a consensus they are empty if primary key or unique key is not set.

Reply:
In current parser, `primaryKeyList` and `uniqueKeysList ` would never be null, they could be empty if there are no definitions.

Reply:
@danny0405  `primaryKeyList` and `uniqueKeysList` are null if they're not defined because we define them in parserImpls.ftl as null by default, which is different with other fields like propertyList and partitionColumns.

@wuchong I kind of misunderstand your comments in jira. I thought we should keep `primaryKeyList` and `uniqueKeysList` 's default value. I will make them consistent with propertyList and partitionColumns.

Reply:
`primaryKeyList` default is an empty list[1], also is `uniqueKeysList`[2], i agree the code is not that intuitive, i would make some refactoring if i have time.

[1] https://github.com/apache/flink/blob/dbe1bfa31db4a561b6faa9c1235f02dc130825ca/flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl#L100
[2]https://github.com/apache/flink/blob/dbe1bfa31db4a561b6faa9c1235f02dc130825ca/flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTable.java#L316

Reply:
@danny0405  Yes, but this only works when it has columns. It doesn't work for sql "create table tbl1". In this PR, I initialize `primaryKeyList` and `uniqueKeysList` to avoid this situation.

Reply:
Well, exactly, the `TableCreationContext` only was initialized when we encounter the `(`. I think we can give a `SqlNodeList.EMPTY` as default value for `primaryKeyList` and `uniqueKeysList` instead of null.

BTW, what is the use case for table without column definitions ? Can you explain a little more ?

Reply:
Thanks for pointing it out. I'm not sure about the use case =_= (Maybe creating an empty table firstly and adding columns into it?). My motivation is that we should make `primaryKeyList` and  `uniqueKeysList`'s default value consistent with other fields like `partitionKeyList`.

Reply:
Currently we did support create empty columns table with our parser, i have no strong objections to add test cases for it.

Reply:
Thanks. Can we resolve this now :) ?

Reply:
No, can we make the `primaryKeyList` and `uniqueKeysList` default as `SqlNodeList.EMPTY` instead of null ? So that in `SqlCreateTable`, we can mark them as 'requireNotNull'.

Reply:
`SqlNodeList primaryKeyList = SqlNodeList.EMPTY;`
`List<SqlNodeList> uniqueKeysList = new ArrayList<SqlNodeList>();`

I've already changed their default value yesterday. Can you take a look ?